src/idl_gen_cpp.cpp

```diff
@@ -199,6 +201,20 @@ class CppGenerator : public BaseGenerator {
     for (auto kw = keywords; *kw; kw++) keywords_.insert(*kw);
   }
 
+  void GenFlatbuffersVersionCheck() {
```

Comment:
It might be confusing why we'd want exact equality instead of the sem-ver compatibility so consider leaving a comment. Otherwise LGTM

Reply:
Done.